code/modules/projectiles/gun_attachables.dm

```diff
@@ -250,6 +260,8 @@ inaccurate. Don't worry if force is ever negative, it won't runtime.
 	master_gun.scatter_unwielded			-= scatter_unwielded_mod
 	master_gun.aim_speed_modifier			-= initial(master_gun.aim_speed_modifier)*aim_mode_movement_mult
 	master_gun.iff_marine_damage_falloff	-= shot_marine_damage_falloff
+	master_gun.aim_fire_delay 				+= cached_aim_mode_debuff_fire_rate
```

Comment:
```suggestion
	master_gun.aim_fire_delay 				+= cached_aim_mode_debuff_fire_rate
	if(CHECK_BITFIELD(master_gun.flags_gun_features, GUN_IS_AIMING))
		master_gun.modify_fire_delay(cached_aim_mode_debuff_fire_rate)
```
This'll catch the current issue where you can reduce a gun's fire delay to zero by removing the RDS while aiming.